test/caf_co_reduce_test.f90

```diff
@@ -39,9 +40,9 @@ function alphabetically_1st_size1_string_array() result(result_)
     end associate
 
     call prif_num_images(num_images=num_imgs)
-    associate(expected_name => minval(names(1:min(num_imgs, size(names)))))
-      result_ = assert_that(all(expected_name == my_name))
-    end associate
+    !expected_name = minval(names(1:min(num_imgs, size(names)))) ! this exposes a flang bug
+    expected_name = "betty"
```

Comment:
Similarly here, I believe this answer assumes `num_imgs >= 4`.

Can we rewrite the commented-out test code to avoid the flang bug while still preserving the appropriate logic?